spring-kafka/src/main/java/org/springframework/kafka/listener/ConsumerAwareThreadStateProcessor.java

```diff
@@ -18,43 +18,34 @@
 
 import org.apache.kafka.clients.consumer.Consumer;
 
+import org.springframework.kafka.support.ThreadStateProcessor;
+
 /**
- * An interceptor for consumer poll operation.
- *
- * @param <K> the key type.
- * @param <V> the value type.
+ * A {@link ThreadStateProcessor} involving a {@link Consumer}.
  *
- * @author Gary Russell
  * @author Karol Dowbecki
- * @author Artem Bilan
+ * @author Gary Russell
  * @since 2.8
  *
  */
-public interface BeforeAfterPollProcessor<K, V> {
+public interface ConsumerAwareThreadStateProcessor extends ThreadStateProcessor {
```

Comment:
From perspective of someone who uses the library and only infrequently digs into implementation it feels like common interfaces are getting quite complex to use. 

If I were to implement a new `RecordInterceptor` which methods exactly do I need to implement? I'd have to review how the `RecordInterceptor` interface is used to spot that only `ConsumerAwareThreadStateProcessor.setupThreadState(Consumer)` is called and `ConsumerAwareThreadStateProcessor.setupThreadState()` is not used,

Perhaps `ConsumerAwareThreadStateProcessor` should not extend `ThreadStateProcessor` if it's not overriding the parent methods? Or maybe it should override the methods and throw `UnsupportedOperationException` telling that the parent methods are not used?

Reply:
Good point; I remove the inheritance.

Reply:
I removed it altogether:

```
Remove ThreadStateProcessor (TSP), rename CATSP to TSP.
- confusing for interceptor implementors
- no components ever call setupThreadState; ARP and CEH build state during normal processing.
```